crates/vault-registry/src/lib.rs

```diff
@@ -612,9 +696,10 @@ impl<T: Config> Pallet<T> {
         // ensure the vault is active
         let _vault = Self::get_active_rich_vault_from_id(vault_id)?;
 
+        // will fail if collateral ceiling exceeded
+        Self::try_increase_total_backing_collateral(amount)?;
         // will fail if free_balance is insufficient
         amount.lock(vault_id)?;
```

Comment:
Also here I am assuming that all our calls are transactional, otherwise e.g. collateral total could be increased but then locking collateral could fail - so just to make sure, does this assumption hold?

Reply:
Yes, all of our _dispatchable_ functions are transactional (through the `#[transactional]` above each function)